### ISSUE TITLE
[BUG] url field jumps out of viewport when pasting a link there; link isn't being added eventually

### ISSUE THREAD

**Issue by goodwin64** [sanity-io/sanity#5909]
**Describe the bug**

The URL field jumps out of the user's viewport when pasting a link there; the link isn't being added eventually.

**To Reproduce**

1. Go to documents, open the section that has a big chunk of text
2. Select the text you want to add a link to
3. Click on the "pencil" icon

**Expected behavior**
You can paste a link in the URL field

**Actual behavior**
URL field jumps out of the section viewport, see the recording below:

**Screenshots**

https://github.com/sanity-io/sanity/assets/10671879/b86a5e42-76fe-41b8-853b-767f6a04169e

**Which versions of Sanity are you using?**

```
@sanity/cli (global)          3.28.0 (latest: 3.30.1)
@sanity/cli                   3.28.0 (latest: 3.30.1)
@sanity/eslint-config-studio   2.0.4 (latest: 3.0.1)
@sanity/icons                 2.10.1 (latest: 2.10.3)
@sanity/ui                     1.9.3 (latest: 2.0.7)
@sanity/vision                3.28.0 (latest: 3.30.1)
sanity                        3.28.0 (latest: 3.30.1)
```

**What operating system are you using?**

MacOS Sonoma 14.1.1 (23B81)

**Which versions of Node.js / npm are you running?**

```
9.7.2
v18.17.1
```

**Additional context**

1. You need to have a big chunk of text - I used first 5 paragraphs of [Lorem Ipsum](https://www.lipsum.com/feed/html).
2. There's a workaround shown in the video - you may use the "enlarge section" button. But you need to do it for each link.

**Security issue?**

No

**Comment by ninaandal** [comment]
Hi @goodwin64 ! Thank you for reporting, we have gotten some more signal of this and  😄 We can reproduce it internally and the bug has been added to triage, we will keep this issue updated once it's fixed.

**Comment by goodwin64** [comment]
Thanks a lot for getting back on this! Looking forward for the fix!

**Comment by ninaandal** [comment]
Hi again! A fix for this was included in v3.40.0 ✨ Thanks again for reporting!
